File: Alisto.Api/Models/IssueReport.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Alisto.Api.Enums;

namespace Alisto.Api.Models
{
    public class IssueReport
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        public string? UserId { get; set; } // Optional for anonymous reports

        [Required]
        [MaxLength(20)]
        public string ReferenceNumber { get; set; }

        public IssueCategory Category { get; set; }

        public UrgencyLevel UrgencyLevel { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        [MaxLength(500)]
        public string Location { get; set; }

        [MaxLength(100)]
        public string? Coordinates { get; set; } // GPS coordinates

        [MaxLength(200)]
        public string? ContactInfo { get; set; }

        public IssueStatus Status { get; set; } = IssueStatus.Submitted;

        public Priority Priority { get; set; }

        [MaxLength(100)]
        public string? AssignedDepartment { get; set; }

        [MaxLength(100)]
        public string? AssignedTo { get; set; }

        public DateTime? EstimatedResolution { get; set; }

        public DateTime? ActualResolution { get; set; }

        public string? ResolutionNotes { get; set; }

        public bool PubliclyVisible { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        [ForeignKey("UserId")]
        public virtual User? User { get; set; }

        public virtual ICollection<IssuePhoto> Photos { get; set; } = new List<IssuePhoto>();
        public virtual ICollection<IssueUpdate> Updates { get; set; } = new List<IssueUpdate>();
    }
}

File: types/issue-report.ts
// Issue report-related types for issue reporting endpoints

import { ApiResponse, PaginatedResponse, IssueCategory, IssueStatus, UrgencyLevel, Priority } from './common';

// ============================================================================
// REQUEST TYPES
// ============================================================================

export interface CreateIssueReportRequest {
  userId?: string; // Optional for anonymous reports
  category: IssueCategory;
  urgencyLevel: UrgencyLevel;
  title: string;
  description: string;
  location: string;
  coordinates?: string; // GPS coordinates
  contactInfo?: string;
  priority: Priority;
  publiclyVisible?: boolean;
}

export interface UpdateIssueStatusRequest {
  status: IssueStatus;
  assignedDepartment?: string;
  assignedTo?: string;
  estimatedResolution?: string; // ISO date string
  resolutionNotes?: string;
}

// ============================================================================
// RESPONSE TYPES
// ============================================================================

export interface IssuePhotoDto {
  id: string;
  photoUrl: string;
  caption?: string;
  uploadedAt: string; // ISO date string
}

export interface IssueReportDto {
  id: string;
  referenceNumber: string;
  category: string; // IssueCategory as string
  urgencyLevel: string; // UrgencyLevel as string
  title: string;
  description: string;
  location: string;
  coordinates?: string;
  status: string; // IssueStatus as string
  priority: string; // Priority as string
  assignedDepartment?: string;
  estimatedResolution?: string; // ISO date string
  createdAt: string; // ISO date string
  photos: IssuePhotoDto[];
}

export interface IssueCategoryDto {
  value: string;
  displayName: string;
}

export interface IssueStatusDto {
  value: string;
  displayName: string;
}

// ============================================================================
// API RESPONSE TYPES
// ============================================================================

export type GetIssueReportsResponse = ApiResponse<IssueReportDto[]>;
export type GetIssueReportResponse = ApiResponse<IssueReportDto>;
export type CreateIssueReportResponse = ApiResponse<IssueReportDto>;
export type UpdateIssueReportResponse = ApiResponse<IssueReportDto>;
export type DeleteIssueReportResponse = ApiResponse<object>;
export type UpdateIssueStatusResponse = ApiResponse<IssueReportDto>;
export type GetIssueCategoriesResponse = ApiResponse<IssueCategoryDto[]>;
export type GetIssueStatusesResponse = ApiResponse<IssueStatusDto[]>;

// ============================================================================
// QUERY PARAMETER TYPES
// ============================================================================

export interface GetIssueReportsQueryParams {
  page?: number;
  pageSize?: number;
  category?: string;
  status?: string;
  priority?: string;
} 